Sources/Apollo/GraphQLOperation.swift

```diff
@@ -6,14 +6,15 @@ public enum GraphQLOperationType {
 
 public protocol GraphQLOperation: class {
   var operationType: GraphQLOperationType { get }
-  
+
   var operationDefinition: String { get }
   var operationIdentifier: String? { get }
-  
+  var operationName: String? { get }
```

Comment:
Should this be nullable?

Reply:
I think with the changes in codegen we *should* be able to make this non-null since it's now auto-generated.

Reply:
I opted for an Optional In case the tooling wasn’t updated (either by the user or within the tooling itself), to soften the dependency

Reply:
fair enough - we can certainly leave it optional for a bit but I think eventually we should make this required.

Reply:
heads up: Had to make this non-nullable because protocol resolution stuff made this always return `nil` even if there was an implemented `operationName`. I can give a longer explanation if desired but it's basically [this example of how protocols with default implementations resolve which implementation to use](https://speakerdeck.com/designatednerd/protocols-all-the-way-down-dotswift-paris-january-2018?slide=70) and why that's not always what you expect.